### PR TITLE
fix(cli): Remove `prisma2` bin

### LINKED ISSUE #19438: Prisma CLI: remove `prisma2` "executable"
Prisma CLI still includes code that it is usable via `prisma2`. One would think that to only be a link or something that needs little to no space, but on MacOS this leads to a 4.5 MB file `node_modules/.bin/prisma2`. 

Usually that does not matter much, but sometimes this [can also be included in serverless functions when not configured properly](https://github.com/prisma/prisma/issues/19358) - making those 4.5MB bigger than they would need to be (and should be, as the CLI of course usually is not needed at all).

We have deprecated the `prisma2` command ages ago, more than 3 years actually with version `2.0.0-beta.3`:
PR: https://github.com/prisma/prisma/pull/2215
https://github.com/prisma/prisma/blob/bfc2193876265c9e36312c661396245dff8a20d0/packages/cli/src/bin.ts#L66-L73

### PR DESCRIPTION
**Pull request by janpio**
closes https://github.com/prisma/prisma/issues/19438

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -56,8 +56,7 @@
     ]
   },
   "bin": {
-    "prisma": "build/index.js",
-    "prisma2": "build/index.js"
+    "prisma": "build/index.js"
```
Comment (janpio) [inline packages/cli/package.json:59]: This is the main change.

File: packages/cli/src/bin.ts [hunk 2]
```diff
@@ -63,15 +63,6 @@ process.once('SIGINT', () => {
...
 })
 
-if (process.argv.length > 1 && process.argv[1].endsWith('prisma2')) {
-  console.log(
-    yellow('deprecated') +
-      `  The ${red('prisma2')} command is deprecated and has been renamed to ${green('prisma')}.\nPlease execute ${bold(
-        green('prisma' + (commandArray.length ? ' ' + commandArray.join(' ') : '')),
-      )} instead.\n`,
-  )
-}
```
Comment (janpio) [inline packages/cli/src/bin.ts:73]: This removes the deprecation warning code.

File: packages/cli/scripts/preinstall.js [hunk 3]
```diff
@@ -24,11 +24,11 @@ function isPackageInstalledGlobally(name) {
...
-  if (prisma2InstalledGlobally) {
+  const prismaInstalledGlobally = isPackageInstalledGlobally('prisma')
+  if (prismaInstalledGlobally) {
     return {
-      ...prisma2InstalledGlobally,
-      name: 'prisma2',
+      ...prismaInstalledGlobally,
+      name: 'prisma',
     }
   }
```
Comment (janpio) [inline packages/cli/scripts/preinstall.js:33]: This logic seems to have relied on the deprecated bin/executable, and can not any more.

Comment (Jolg42) [reply packages/cli/scripts/preinstall.js:33]: Looks to me that the whole `prismaIsInstalledGlobally()` function and usage could go away as later in this file it's only used to check if `prisma2` is installed globally and error with a message (see line 62+)

Comment (janpio) [reply packages/cli/scripts/preinstall.js:33]: Oh you are right, my change breaks that functionality.

I had considered removing that bit as well, but it is not about the `prisma2` bin of the current Npm package, but about a `prisma2` npm package we used to have. With us not having the`prisma2` bin any more, we honestly don't care if someone still has that old Npm package installed I think.

Comment (janpio) [reply packages/cli/scripts/preinstall.js:33]: This feels pretty drastic, but along those lines?
https://github.com/prisma/prisma/pull/19439

Comment (Jolg42) [reply packages/cli/scripts/preinstall.js:33]: Yes that's it, I think this removal is due and the clean slate very nice

File: packages/cli/src/bin.ts (hunk 2 above)
Comment (Jolg42) [reply packages/cli/src/bin.ts:73]: I removed the green and yellow imports, which I found in the linting errors as unused

File: packages/cli/scripts/preinstall.js
Comment (janpio) [inline packages/cli/scripts/preinstall.js:1]: This change completely removes handling of a `prisma2` Npm package or bin being available on the system.
(Also allowed us to get rid of `is-installed-globally`)

File: packages/cli/scripts/preinstall.js (hunk 3 above)
Comment (janpio) [reply packages/cli/scripts/preinstall.js]: Merged and also removed dependency.

**Review by Jolg42 — APPROVED** [review]

**Pull request merged**